src/utilities/graphql/storeUtils.ts

```diff
@@ -194,6 +195,11 @@ const KNOWN_DIRECTIVES: string[] = [
   "nonreactive",
 ];
 
+// Default stable JSON.stringify implementation used by getStoreKeyName. Can be
+// updated/replaced with something better by calling
+// getStoreKeyName.setStringify(newStringifyFunction).
+let storeKeyNameStringify = canonicalStringify;
```

Comment:
```suggestion
let storeKeyNameStringify: (value: any) => string = canonicalStringify;
```

Otherwise, `setStringify` can only be called with `stringify` functions that share the signature of `canonicalStringify` - which means they all would need to come with a `.reset` method.